workflow/metrics/metrics_k8s_request.go

```diff
@@ -39,7 +48,9 @@ type metricsRoundTripper struct {
 var k8sMetrics metricsRoundTripper
 
 func (m metricsRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	startTime := time.Now()
 	x, err := m.roundTripper.RoundTrip(r)
+	duration := time.Since(startTime)
```

Comment:
it is not safe to assume monotonic time here, the duration value could be negative.
I would suggest dropping a negative `duration`, but will leave the decision up to you.

Reply:
My understanding is that this should be robust (on linux where a monotonic clock will exist).
https://pkg.go.dev/time#hdr-Monotonic_Clocks

Reply:
My problem with adding the "don't record below zero" is that it:
* Silently breaks this as a correct counter
* Acknowledges that the result is sometimes a lie. If it's sometimes a lie we can't tell when.

I don't think this happens, so I'm going to go with it as is.